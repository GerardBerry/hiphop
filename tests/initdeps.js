"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module B A=${{initApply: function() {return this.value.B}}}>
        <hh.emit B value=${5}/>
	<hh.emit A/>
	<hh.pause/>
	<hh.let Y>
	  <hh.let X=${{initApply: function() {return this.value.Y}}}>
            <hh.parallel>
	      <hh.emit X/>
	      <hh.emit Y value=${55}/>
	    </hh.parallel>
	    <hh.emit A apply=${function() {return this.value.X}}/>
	  </hh.let>
	</hh.let>
      </hh.module>

const m = new hh.ReactiveMachine(prg);

console.log(m.react());
console.log(m.react());
