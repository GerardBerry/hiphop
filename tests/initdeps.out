[ 'B(5)', 'A(5)' ]
[ 'A(55)' ]
